datalad/interface/annotate_paths.py

```diff
@@ -546,9 +546,25 @@ def __call__(
                         **dict(res_kwargs, **path_props))
                     res['status'] = nondataset_path_status
                     res['message'] = 'path not associated with reference dataset'
+                    # TODO: That `path` in following line seems to be incorrect;
+                    # we are operating on `r` or `p` respectively. Guess it
+                    # should be `r`.
+                    # Figure out, whether or not to report on the resolved one:
                     reported_paths[path] = res
                     yield res
 
+            # preserve non-existing paths to be silently killed by modification
+            # detection and append them to requested_paths again after detection.
+            # TODO: This might be melted in with treatment of non dataset paths
+            # above. Re-appending those paths seems to be better than yielding
+            # directly to avoid code duplication, since both cases later on are
+            # dealt with again.
+            preserved_paths = []
+            if requested_paths:
+                for r in requested_paths:
+                    if not lexists(r['path'] if isinstance(r, dict) else r):
+                        preserved_paths.append(r)
+
```

Comment:
I aggree with the approach, but this should be a simple list comprehension, no?

Reply:
Right. Will change.